### PR TITLE
User enters in new reminder and it renders to page 

### PR DESCRIPTION
**Pull request by gness1804**
## Purpose

_Describe the problem or feature in addition to a link to the issues. Delete any content that doesn't apply._

The user can now enter a new reminder, which renders to the page on submit. 

Closes #4 

## Approach

_How does this change address the problem?_

This change now makes the page interaction, since the user can now add a new reminder, rather than just viewing reminders. 

### Learning

_Describe the research stage._

_Links to blog posts, patterns, libraries or add-ons used to solve this problem._

The research for this step involved consulting other students and trial-and-error. 

### Test coverage 

_Briefly describe what tests you have written and why._

We wrote two acceptance tests which test that the page route to the correct URL when user clicks the add reminder button; input field render when the user clicks this button; and the user can enter information that then is correctly rendered to the page. We also wrote an integration test that checks that the name value for the component is updated with the correct value. These tests are important to ensure that the add reminders functionality works properly.

### CODE REVIEW

File: app/routes/reminders/new.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+import Ember from 'ember';
```
Comment (martensonbj) [inline app/routes/reminders/new.js]: Looks like we aren't using this file. Let's get rid of it.

File: app/templates/components/new-reminder-form.hbs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  </label>
+  <label>
+    Date
+    {{input class="add-date-input" value=date placeholder="Date..."}}
+  </label>
+  <label>
+    Notes
+    {{textarea class="add-notes-input" value=notes placeholder="Notes..."}}
+  </label>
+  {{!-- <button class="create-reminder-button">Save</button> --}}
```
Comment (martensonbj) [inline app/templates/components/new-reminder-form.hbs]: Please remove comments before submitting a PR 😄

File: tests/acceptance/add-new-reminder-test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  visit('/reminders/');
+  click('.add-reminder-button')
+
+  andThen(function() {
+    assert.equal(currentURL(), '/reminders/new');
+    assert.equal(find('.create-new--form').length, 1);
+    assert.equal(find('.add-title-input').length, 1);
+    assert.equal(find('.add-date-input').length, 1);
+    assert.equal(find('.add-notes-input').length, 1);
+  });
```
Comment (martensonbj) [inline tests/acceptance/add-new-reminder-test.js]: Love the thorough testing ! 👍

File: tests/acceptance/reminder-list-test.js [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
 
 import Ember from 'ember';
 
 moduleForAcceptance('Acceptance | reminders list');
 
-test('viewing the homepage', function(assert) {
+test('viewing the homepage re-routes to /reminders, rendering 5 reminders', function(assert) {
   server.createList('reminder', 5);
 
   visit('/');
```
Comment (martensonbj) [inline tests/acceptance/reminder-list-test.js]: Let's finish up this test - assert the correct route shows up and you see the reminders you expect.

File: tests/acceptance/add-new-reminder-test.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+test('clicking "Add a New Reminder" will render a form on the page to add new reminders', function(assert) {
+  visit('/reminders/');
+  click('.add-reminder-button')
+
+  andThen(function() {
+    assert.equal(currentURL(), '/reminders/new');
+    assert.equal(find('.create-new--form').length, 1);
+    assert.equal(find('.add-title-input').length, 1);
+    assert.equal(find('.add-date-input').length, 1);
+    assert.equal(find('.add-notes-input').length, 1);
```
Comment (martensonbj) [inline tests/acceptance/add-new-reminder-test.js]: Dig the thorough testing! 👍

File: tests/unit/routes/new-test.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { moduleFor, test } from 'ember-qunit';
```
Comment (martensonbj) [inline tests/unit/routes/new-test.js]: Won't be needing this test - you can delete the file.

File: tests/acceptance/reminder-list-test.js (hunk 4 above)
Comment (martensonbj) [inline tests/acceptance/reminder-list-test.js]: Ignore this - test is fine.

**Comment by gness1804** [comment]
These changes add the revisions requested in the pull request.

File: app/templates/reminders.hbs [hunk 7]
```diff
@@ -2,13 +2,19 @@
...
       {{/link-to}}
+
       <span class="spec-reminder-date">{{reminder.date}}</span>
     </li>
   {{/each}}
 </ul>
+{{#link-to 'reminders.new'}}
+  <button class="add-reminder-button">Add a New Reminder</button>
+  {{!-- <button class="add-reminder-button" {{action "createReminder"}}>Add a New Reminder</button> --}}
+  {{!-- Create a New Reminder --}}
```
Comment (Tman22) [inline app/templates/reminders.hbs]: Will you delete this committed out code?

**Pull request merged**